apps/server/src/modules/learnroom/mapper/rolename.mapper.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { RoleName } from '@shared/domain/interface';
+import { UserAndAccountTestFactory, courseFactory, roleFactory, setupEntities, userFactory } from '@shared/testing';
+import { RoleNameMapper } from './rolename.mapper';
+
+describe('rolename mapper', () => {
+	let module: TestingModule;
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	beforeAll(async () => {
+		await setupEntities();
+		module = await Test.createTestingModule({
+			imports: [],
+			providers: [],
+		}).compile();
+	});
+
+	const setup = () => {
+		const { teacherUser } = UserAndAccountTestFactory.buildTeacher({}, []);
+		const { studentUser } = UserAndAccountTestFactory.buildStudent({}, []);
```

Comment:
Why not use `userFactory`?  `UserAndAccountTestFactory` is more appropriate for API tests. You can even use `userFactory.asTeacher()` or `userFactory.asStudent()` when needed.